packages/container-registry-v5/lib/sanbashi.js

```diff
@@ -151,6 +151,14 @@ Sanbashi.cmd = function (cmd, args, options = {}) {
         stdout += data.toString()
       })
     }
+    child.on('error', (err) => {
+      if (err.code === 'ENOENT' && err.path === 'docker') {
+        reject(new Error(`Cannot find docker, please ensure docker is installed.
+        If you need help installing docker, visit https://docs.docker.com/install/#supported-platforms`))
+      } else {
+        reject(err)
+      }
+    })
     child.on('exit', (code, signal) => {
       if (signal || code) reject(signal || code)
```

Comment:
Thinking out loud...

Here we're handling when the spawned process finishes, but does so with an non-zero (success) exit code.  This will happen if the `spawned` process is executed, but fails during execution.

In the `on('error')` we're handing failures that occur before the `spawned` process has a chance to execute e.g. `ENOENT` error.  Obviously, there are other pre-spawn errors that can occur, so would it suffice to just `reject` those errors as is?

Reply:
not sure I fully understand this, I think the most recent commit you have to this PR should handle every situation as this is already handling non-zero status (though I think it isn't handling it with the clearest error, it's just going to say the exit status number)

Reply:
@jdxcode correct.  Ok, despite my rambling, I believe we're on the same page.